File: runtime/CSlib/optional.cs
namespace bigloo 
{
  public sealed class optional: cnst 
  {
    public static readonly optional _optional= new optional( 0x102 );

    private optional( int  n )
      : base( n )
    {
    }

    public override void write( output_port  p ) 
    {
      p.write( "#!optional" );
    }
  }
}

File: runtime/Jlib/optional.java
package bigloo;

public final class optional extends cnst
{
  public static final optional optional= new optional( 0x102 );

  private optional( final int  n )
  {
    super( n );
  }

  public void write( final output_port  p )
  {
    p.write( "#!optional" );
  }
}